src/Microsoft.AspNet.Authentication/Events/SigningInContext.cs

```diff
@@ -25,6 +28,13 @@ public abstract class ReturnEndpointContext : EndpointContext
         public ClaimsPrincipal Principal { get; set; }
         public AuthenticationProperties Properties { get; set; }
 
+        public bool IsRequestCompleted { get; private set; }
+
+        public void RequestCompleted()
+        {
+            IsRequestCompleted = true;
```

Comment:
Can we change the name of this method to have a nice verb in it then? Or do we really need this method at all?